src/chronify/time_configs.py

```diff
@@ -159,47 +158,137 @@ class AnnualTimeRange(TimeBaseModel):
     def list_time_columns(self) -> list[str]:
         return [self.time_column]
 
+    def list_time_zone_column(self) -> list[str]:
+        return []
 
-class IndexTimeRange(TimeBaseModel):
-    time_type: Literal[TimeType.INDEX] = TimeType.INDEX
-    start: int
+
+class IndexTimeRangeBase(TimeBaseModel):
+    """Defines a time range in the form of indexes"""
+
+    time_column: str = Field(description="Column in the table that represents index time.")
+    start: int = Field(description="starting index")
     length: int
-    resolution: timedelta
-    time_zone: TimeZone
+    start_timestamp: datetime = Field(
+        description="The timestamp represented by the starting index."
+    )
+    resolution: timedelta = Field(description="The resolution of time represented by the indexes.")
+    time_zone_column: Optional[str] = Field(
```

Comment:
I'll suggest a solution that seems cleaner, at least to me. Include `time_zone_column` in models as a required parameter where required. Don't include it in models that won't use it. Implement this default method in this base class:
```python
def get_time_zone_column(self) -> str | None:
    return None
```
Override that method in classes where the field is required and return that value.